main.c

```diff
@@ -0,0 +1,229 @@
+    struct User user;
+    struct User users[100]; // Array to store users
+    int userCount = 0;
+    FILE *file = fopen(FILENAME, "r"); // Open file for reading
+
+    if (!file) {
+        printf("Error opening file!\n");
+        return;
+    }
+
+    printf("Enter User ID to update: ");
+    scanf("%d", &id);
+    getchar();  // Clear the newline left by scanf
+
+    // Read all users into the array
+    while (fscanf(file, "%d,%49[^,],%d\n", &user.id, user.name, &user.age) != EOF) {
+        users[userCount++] = user;
+    }
+
+    fclose(file); // Close file after reading
+
+    int found = 0;
+    // Find and update the user with the matching ID
+    for (int i = 0; i < userCount; i++) {
+        if (users[i].id == id) {
+            found = 1;
+            printf("User found! Current details: %d, %s, %d\n", users[i].id, users[i].name, users[i].age);
+
+            // Get new details
+            printf("Enter new Name: ");
+            fgets(users[i].name, sizeof(users[i].name), stdin);
+            users[i].name[strcspn(users[i].name, "\n")] = '\0'; // Remove newline character
+            printf("Enter new Age: ");
+            scanf("%d", &users[i].age);
+
+            printf("User updated successfully.\n");
+            break;
+        }
+    }
+
+    if (!found) {
+        printf("User with ID %d not found.\n", id);
+        return;
+    }
+
+    // Open file for writing (overwrite mode)
+    file = fopen(FILENAME, "w"); // Open file to overwrite the content
+
+    if (!file) {
+        printf("Error opening file!\n");
+        return;
+    }
+
+    // Write all users back to the file, including the updated user
+    for (int i = 0; i < userCount; i++) {
+        fprintf(file, "%d,%s,%d\n", users[i].id, users[i].name, users[i].age);
+    }
+
+    fclose(file); // Close the file after writing
+}
+
+// Function to delete a user by ID
+void deleteUser() {
+    int id;
+    struct User user;
+    FILE *file = fopen(FILENAME, "r");
+
+    if (!file) {
+        printf("Error opening file!\n");
+        return;
+    }
+
+    FILE *tempFile = fopen("temp.txt", "w"); // Temporary file to store updated records
+
+    if (!tempFile) {
+        printf("Error opening temporary file!\n");
+        fclose(file);
+        return;
+    }
+
+    printf("Enter User ID to delete: ");
+    scanf("%d", &id);
+
+    int found = 0;
+    while (fscanf(file, "%d,%49[^,],%d\n", &user.id, user.name, &user.age) != EOF) {
+        if (user.id == id) {
+            found = 1;
+            printf("User with ID %d has been deleted.\n", id);
+        } else {
+            // Copy user to temp file if not deleted
+            fprintf(tempFile, "%d,%s,%d\n", user.id, user.name, user.age);
+        }
+    }
+
+    if (!found) {
+        printf("User with ID %d not found.\n", id);
+    }
+
+    fclose(file);
+    fclose(tempFile);
+
+    // Remove original file and rename temp file to original name
+    remove(FILENAME);
+    rename("temp.txt", FILENAME);

```

Comment:
Use only one txt file. It's not necessary to maintain the user details after the program termination.